### ISSUE TITLE
Upload of WebJob fails with Azure CLI on Linux (process out of memory)

### ISSUE THREAD

**Issue by oyvindholmstad** [Azure/azure-xplat-cli#2173]
We are trying to use the Azure CLI on linux to upload a WebJob as part of our continuous deployment pipeline.

> azure site job upload -v $WEB_JOB_NAME $WEB_JOB_TYPE run.zip $WEB_SITE_NAME

But the command fails after > 20 mins of waiting on the "Uploading WebJob" step.

> FATAL ERROR: CALL_AND_RETRY_2 Allocation failed - process out of memory

Some more info:

- The cli is properly authenticated. We can trigger already existing WebJobs just fine.
- The exact same run.zip uploads successfully from Microsoft Azure Powershell on Windows.
- The zip-file contains a runnable jar, and a small .cmd-script to start it. File size: 30 MB
- We tried setting the verbose-flag, but it does not give any more information.

**Comment by delgadom** [comment]
I am getting the same error when downloading a large file through the node.js interface.

    <--- Last few GCs --->

      761431 ms: Mark-sweep 1394.8 (1445.7) -> 1394.7 (1445.7) MB, 442.9 / 0 ms [allocation failure] [GC in old space requested].
      761869 ms: Mark-sweep 1394.7 (1445.7) -> 1394.7 (1445.7) MB, 439.0 / 0 ms [allocation failure] [GC in old space requested].
      762324 ms: Mark-sweep 1394.7 (1445.7) -> 1391.4 (1445.7) MB, 455.2 / 0 ms [last resort gc].
      762760 ms: Mark-sweep 1391.4 (1445.7) -> 1394.7 (1445.7) MB, 434.8 / 0 ms [last resort gc].


    <--- JS stacktrace --->

    ==== JS stack trace =========================================

    Security context: 0000017DB40E3AD1 <JS Object>
        1: emitNode(aka emitNode) [C:\Users\#######\node_modules\azure-storage\node_modules\xml2js\node_modules\sax\lib\sax.js:~592] [pc=0000023443B9414E] (this=0000017DB4004189 <undefined>,parser=0000012B3030C751 <an SAXParser with map 000003076F542481>,nodeType=0000012B30314519 <String[10]: onclosetag>,data=0000024FCBF93989 <String[16]: Content-Encoding>)
        2...

    FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory

Did you ever find a solution?

**Comment by oyvindholmstad** [comment]
No, I ended up using the [REST Api](https://github.com/projectkudu/kudu/wiki/WebJobs-API). This is from my upload script:

```
# Workaround cause upload does not work from CLI

siteScmUrl="https://mywebsite.scm.azurewebsites.net/"
publishingUserName='user'
publishingPassword='password'
jobPath="api/triggeredwebjobs/$WEB_JOB_NAME"
fullUrl=$siteScmUrl$jobPath

# Upload the zip file using curl 
curl -v -X PUT --data-binary @run.zip -u $publishingUserName:$publishingPassword -H 'Content-Type: application/zip' -H 'Content-Disposition: attachment; filename="run.cmd"' $fullUrl
```

Nb! I start the webjob with run.cmd, so you may have to remove that if your zip has something else.

**Comment by christopheranderson** [comment]
This is likely an issue with the Azure CLI itself, and not a service side issue. Using the REST API is definitely the best course of action, for now, if you hit this issue.

**Comment by tfmorris** [comment]
Too bad no one from MSFT care about this after a year! I'm running into the same issue attempting to use the pre-built Azure CLI Docker image.

`docker run microsoft/azure-cli azure storage blob list`

after peeling a few layers of the onion, the workaround that I'm currently attempt to use is:

`docker run microsoft/azure-cli node --max_old_space_size=8000000 /usr/bin/azure <command>`

http://stackoverflow.com/questions/26094420/fatal-error-call-and-retry-last-allocation-failed-process-out-of-memory

That fixes the OOM error, but runs into a "invalid string length" error, so one step closer.

**Comment by amarzavery** [comment]
@ahmedelnably - Can you take a look at this issue?
